russstation/code/game/machinery/compost.dm

```diff
@@ -0,0 +1,55 @@
+/obj/machinery/portable_atmospherics/canister/compost_bin
+	name = "compost bin"
+	desc = "A compost bin that decomposes grown food into miasma."
+	icon = 'russstation/icons/obj/atmospherics/canisters.dmi'
+	icon_state = "compost"
+	anchored = FALSE
+	volume = 500
+	armor = list(MELEE = 10, BULLET = 0, LASER = 0, ENERGY = 0, BOMB = 0, BIO = 100, RAD = 100, FIRE = 10, ACID = 0)
+	max_integrity = 100
+	canister_overlay_file = null //no overlay
+	gas_type = /datum/gas/miasma
+	filled = 0
+
+/obj/machinery/portable_atmospherics/canister/compost_bin/Initialize()
+	//Also can contain compost material that is helpful to botanists (so theres an insentive to use them other then the fact that it creates miasma gas)
+	create_reagents(300, DRAINABLE)
+	. = ..()
+
+	//these are needed here because canisters override icons on init to default values, this fixes it
+	icon = 'russstation/icons/obj/atmospherics/canisters.dmi'
+	icon_state = "compost"
+	update_icon_state()
+
+/obj/machinery/portable_atmospherics/canister/compost_bin/attackby(obj/item/W, mob/user, params)
+	if(istype(W, /obj/item/food/grown))
+		user.visible_message("[user] puts \the [W.name] into \the [src.name].", "You add \the [W.name] to \the [src.name].", "<span class='hear'>You hear a plop sound.</span>")
```

Comment:
I would assume that the visible messages are meant to describe sight, not sound. If a sound is desired, wouldn't it make sense to play an actual sound? But beyond that, if the player was deafened, wouldn't they still get the text that said they heard something?

Reply:
The third argument is meant for blind people